### ISSUE TITLE
Filtering null records through many to many relation

### ISSUE THREAD

**Issue by empytree** [cycle/orm#143]
Schema as follows (there is nothing interesting about entities and mappers):

```php
'article' => [
    Schema::ENTITY => Article::class,
    Schema::MAPPER => TimestampedMapper::class,
    Schema::DATABASE => 'default',
    Schema::TABLE => 'article',
    Schema::PRIMARY_KEY => 'id',
    Schema::COLUMNS => [
        'id' => 'id',
        'name' => 'source',
        'body' => 'body',
        'created_at' => 'created_at',
        'updated_at' => 'updated_at'
    ],
    Schema::TYPECAST => [
        'id' => 'int',
        'created_at' => 'int',
        'updated_at' => 'int'
    ],
    Schema::RELATIONS => [
        'tags' => [
            Relation::TYPE => Relation::MANY_TO_MANY,
            Relation::TARGET => Tag::class,
            Relation::SCHEMA => [
                Relation::CASCADE => true,
                Relation::THROUGH_ENTITY => ArticleTag::class,
                Relation::INNER_KEY => 'id',
                Relation::OUTER_KEY => 'id',
                Relation::THROUGH_INNER_KEY => 'article_id',
                Relation::THROUGH_OUTER_KEY => 'tag_id',
            ]
        ]
    ]
],
'tag' => [
    Schema::ENTITY => Tag::class,
    Schema::MAPPER => TimestampedMapper::class,
    Schema::DATABASE => 'default',
    Schema::TABLE => 'tag',
    Schema::PRIMARY_KEY => 'id',
    Schema::COLUMNS => [
        'id' => 'id',
        'name' => 'name',
        'created_at' => 'created_at',
        'updated_at' => 'updated_at',
    ],
    Schema::TYPECAST => [
        'id' => 'int',
        'updated_at' => 'int',
        'created_at' => 'int',
    ],
    Schema::RELATIONS => []
],
'article_tag' => [
    Schema::ENTITY => ArticleTag::class,
    Schema::MAPPER => Mapper::class,
    Schema::DATABASE => 'default',
    Schema::TABLE => 'article_tag',
    Schema::PRIMARY_KEY => 'id',
    Schema::COLUMNS => [
        'id' => 'id',
        'article_id' => 'article_id',
        'tag_id' => 'tag_id'
    ],
    Schema::TYPECAST => [
        'id' => 'int',
        'article_id' => 'int',
        'tag_id' => 'int',
    ],
    Schema::RELATIONS => []
]
```

I need to query **only articles with no tags set**. Obviously I need left joining, so I write:

```php
$select = $this->orm->getRepository(Article::class)->select();
$select->with('tags', ['method' => JoinableLoader::LEFT_JOIN]);
$select->where('tags.id', null);
```

Using `sqlStatement()` method I get resulting query:
```sql
SELECT [a lot of stuff]
FROM "article" AS "article" 
LEFT JOIN "article_tag" AS "article_tags_pivot"
    ON "article_tags_pivot"."article_id" = "article"."id" 
INNER JOIN "tag" AS "article_tags"
    ON "article_tags"."id" = "article_tags_pivot"."tag_id"  
WHERE "article_tags"."id" IS NULL
```

Notice the INNER JOIN on the target table. The query gets no results. It should be a LEFT JOIN, which works like a charm.

I have found that second join is hard-coded to be inner here: https://github.com/cycle/orm/blob/master/src/Select/Loader/ManyToManyLoader.php#L139

Questions are:
1. Why?
2. Is there another prescribed way to build the solving query?

**Comment by wolfy-j** [comment]
It has been a bug. Fixed in 2.1.17

**Comment by empytree** [comment]
Heartful thanks for a prompt attention!
